File: ex150.java
import java.util.*;

class Solution {
    public int evalRPN(String[] tokens) {
        Stack<String> stack = new Stack<>();
        
        for (String s : tokens)
            if (!(s.length() == 1 && s.equals("+") || s.equals("-") || s.equals("*") || s.equals("/")))
                stack.push(s);
            else
                if (s.equals("+"))
                    stack.push(Integer.toString(Integer.parseInt(stack.pop()) + Integer.parseInt(stack.pop())));
                else if (s.equals("-"))
                    stack.push(Integer.toString(-(Integer.parseInt(stack.pop()) - Integer.parseInt(stack.pop()))));
                else if (s.equals("*"))
                    stack.push(Integer.toString(Integer.parseInt(stack.pop()) * Integer.parseInt(stack.pop())));
                else if (s.equals("/")) {
                    int b = Integer.parseInt(stack.pop());
                    int a = Integer.parseInt(stack.pop());
                    stack.push(Integer.toString(a / b));
                }

        
        return Integer.parseInt(stack.pop());
    }
}

File: code/ex150.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        vector<string> signs{"+", "-", "*", "/"};
        unordered_set<string> set(signs.begin(), signs.end());
        stack<int> stk;
        for (string s : tokens) {            
            if (set.count(s) > 0) {
                int p2 = stk.top();    stk.pop();
                int p1 = stk.top();    stk.pop();
                if (s == "*")    stk.push(p1 * p2);
                if (s == "/")    stk.push(p1 / p2);
                if (s == "+")    stk.push(p1 + p2);
                if (s == "-")    stk.push(p1 - p2);
            } else {
                stk.push(stoi(s));
            }
        }
        return stk.top();
    }
};